libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SyncState.java

```diff
@@ -82,19 +84,20 @@
 	 * @param store
 	 */
 	public static void setupSyncsSoupIfNeeded(SmartStore store) {
-    	if (store.hasSoup(SYNCS_SOUP) && store.getSoupIndexSpecs(SYNCS_SOUP).length == 2) {
+    	if (store.hasSoup(SYNCS_SOUP) && store.getSoupIndexSpecs(SYNCS_SOUP).length == 3) {
 			return;
 		}
 
 		final IndexSpec[] indexSpecs = {
-			new IndexSpec(SYNC_TYPE, SmartStore.Type.string),
-			new IndexSpec(SYNC_NAME, SmartStore.Type.string)
+			new IndexSpec(SYNC_TYPE, SmartStore.Type.json1),
+			new IndexSpec(SYNC_NAME, SmartStore.Type.json1),
+			new IndexSpec(SYNC_STATUS, SmartStore.Type.json1)
 		};
 
 		// Syncs soup exists but doesn't have all the required indexes
 		if (store.hasSoup(SYNCS_SOUP)) {
 			try {
-				store.alterSoup(SYNCS_SOUP, indexSpecs, false);
+				store.alterSoup(SYNCS_SOUP, indexSpecs, true /* reindexing to json1 is quick */);
```

Comment:
reIndexing when going to json1 is really fast since it's just a bunch of create index calls instead of having to read all rows back to memory / extracting indexed paths / saving back to db.